devito/propagator.py

```diff
@@ -34,6 +34,7 @@ class Propagator(object):
     def __init__(self, name, nt, shape, spc_border=0, time_order=0,
                  time_dim=None, space_dims=None, forward=True, compiler=None,
                  profile=False, cache_blocking=False, block_size=5):
+        self.dtype = np.float32  # set the default precision to single
```

Comment:
There is already a default value set in the constructor for `Operator`. Shouldn't we be sticking to one place for defining defaults?